packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -83,6 +91,54 @@ class TestCommand extends FlutterCommand {
     }
   }
 
+  Future<bool> _collectCoverageData(CoverageCollector collector, { bool mergeCoverageData: false }) async {
+    Status status = logger.startProgress('Collecting coverage information...');
+    String coverageData = await collector.finalizeCoverage();
+    status.stop(showElapsedTime: true);
+
+    String coveragePath = argResults['coverage-path'];
+    File coverageFile = new File(coveragePath)
+      ..createSync(recursive: true)
+      ..writeAsStringSync(coverageData, flush: true);
+    printTrace('wrote coverage data to $coveragePath (size=${coverageData.length})');
+
+    String baseCoverageData = 'coverage/lcov.base.info';
+    if (mergeCoverageData) {
+      if (!os.isLinux) {
+        printError('Merging coverage data is supported only on Linux.');
+        return false;
+      }
+
+      if (!FileSystemEntity.isFileSync(baseCoverageData)) {
+        printError('Missing "$baseCoverageData". Unable to merge coverage data.');
```

Comment:
suggest running update-packages if you detect that the test is in the flutter repo